### PR TITLE
Migration/demo app

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🔗 Issue Links

_Provide all JIRA tickets and/or GitHub issues related to this PR, if applicable._

### 🎯 Goal

_Describe why we are making this change._

### 📝 Summary

_Provide bullet points with the most important changes in the codebase._

### 🛠 Implementation

_Provide a detailed description of the implementation and explain your decisions if you find them relevant._

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

|  Before  |  After  |
| -------- | ------- |
|  `img`   |  `img`  |

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: DemoApp/Sources/Components/AppState.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/AppState.swift:85]: Not a fan of capitalized variables, `voipPushToken` is descriptive enough.

File: DemoApp/Sources/Components/AppEnvironment.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/AppEnvironment.swift:1]: Stream header.

File: DemoApp/Sources/Components/AudioFilter/RobotVoiceFilter.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/AudioFilter/RobotVoiceFilter.swift:2]: Stream header.

File: DemoApp/Sources/Components/Authentication/TokenProvider.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/Authentication/TokenProvider.swift:2]: Stream header.

File: DemoApp/Sources/Components/Authentication/UserProvider.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/Authentication/UserProvider.swift:2]: Same here :)

File: DemoApp/Sources/Components/MicrophoneChecking/MicrophoneChecking.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/MicrophoneChecking/MicrophoneChecking.swift:11]: Why do we need this?

File: DemoApp/Sources/Components/Reactions/ReactionsHelper.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/Reactions/ReactionsHelper.swift:104]: shouldn't we reverse the reaction here?

File: DemoApp/Sources/Components/Services/VoIPPushService.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Components/Services/VoIPPushService.swift:9]: Did you check if the callkit docs need an update with these changes?

File: DemoApp/Sources/Extensions/User+Demo.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Extensions/User+Demo.swift:31]: We can remove Filip and Sam

File: DemoApp/Sources/Extensions/User+Demo.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Extensions/User+Demo.swift:48]: Nice, long overdue 😄 For consistency:
https://getstream.io/static/62cdddcc7759dc8c3ba5b1f67153658c/802d2/ilias-pavlidakis.webp

File: DemoApp/Sources/Views/Controls/DemoControls.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Controls/DemoControls.swift:36]: Feels like too much extra spacing. Off topic, we need to define code conventions for SwiftUI.

File: DemoApp/Sources/Views/Login/LoginViewModel.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Login/LoginViewModel.swift:16]: Why's this commented out?

File: DemoApp/Sources/Views/Login/LoginViewModel.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Login/LoginViewModel.swift:29]: Same here

**Review by martinmitrevski — COMMENTED** [review]
Great work, love what you did with the demo app structure 👏 
Before merging:
- most of the headers are wrong, should have the Stream copyright.
- check if some docs need updating (e.g. callkit)
- small things (see comments)

File: DemoApp/Sources/Components/MicrophoneChecking/MicrophoneChecking.swift
Comment (ipavlidakis) [reply DemoApp/Sources/Components/MicrophoneChecking/MicrophoneChecking.swift:8]: It's not needed any more. It was used when we had the slim version of the SwiftUI SDK

File: DemoApp/Sources/Components/Reactions/ReactionsHelper.swift
Comment (ipavlidakis) [reply DemoApp/Sources/Components/Reactions/ReactionsHelper.swift:101]: The `lowerHand` reaction never leaves the device. On the API call we map it to a raise hand reaction. This is to achieve consistent communication with the other clients. We will never receive a `lowerHand` reaction. It will be a `raiseHand`

File: DemoApp/Sources/Views/Controls/DemoControls.swift
Comment (ipavlidakis) [reply DemoApp/Sources/Views/Controls/DemoControls.swift:37]: I did change that a little bit

File: DemoApp/Sources/Views/Controls/DemoControls.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Controls/DemoControls.swift:149]: Why is this one not used from the defaults?

File: DemoApp/Sources/Views/Controls/DemoControls.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Controls/DemoControls.swift:179]: Same here and the next two.

File: DemoApp/Sources/Views/Reactions/DemoReactionsOverlayView.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/Reactions/DemoReactionsOverlayView.swift:28]: -29?

File: DemoApp/Sources/Views/ReadableContentGuide/ReadableContentGuide.swift
Comment (martinmitrevski) [inline DemoApp/Sources/Views/ReadableContentGuide/ReadableContentGuide.swift:8]: What's this used for?

File: Sources/StreamVideoSwiftUI/CallingViews/CallingGroupView.swift
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/CallingViews/CallingGroupView.swift:134]: Why is this needed?

File: Sources/StreamVideoSwiftUI/Colors.swift
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/Colors.swift:22]: I think this will break few of our demo projects. Best to keep it consistent with the chat colors (which are also UIColors). You could write a computed variable though.

**Review by martinmitrevski — COMMENTED** [review]
Almost there. Few small things, and mostly interested why don't we reuse some call controls from the SDK.

File: Sources/StreamVideoSwiftUI/Colors.swift
Comment (ipavlidakis) [reply Sources/StreamVideoSwiftUI/Colors.swift:22]: What's the point of keeping those as UIColors when the rest are Colors?

File: Sources/StreamVideoSwiftUI/CallingViews/CallingGroupView.swift
Comment (ipavlidakis) [reply Sources/StreamVideoSwiftUI/CallingViews/CallingGroupView.swift:134]: Without it the text inside the participant bubble isn't sized correctly.

| Before  | After |
| ------------- | ------------- |
| Simulator Screenshot - iPhone 14 Pro - 2023-08-31 at 18 08 55 | Simulator Screenshot - iPhone 14 Pro - 2023-08-31 at 18 10 09 |

File: DemoApp/Sources/Views/Reactions/DemoReactionsOverlayView.swift
Comment (ipavlidakis) [reply DemoApp/Sources/Views/Reactions/DemoReactionsOverlayView.swift:28]: To avoid covering the connection status view

File: DemoApp/Sources/Views/Controls/DemoControls.swift
Comment (ipavlidakis) [reply DemoApp/Sources/Views/Controls/DemoControls.swift:149]: No reason, just a remnant from the previous repo. Replacing them

File: DemoApp/Sources/Views/ReadableContentGuide/ReadableContentGuide.swift
Comment (martinmitrevski) [reply DemoApp/Sources/Views/ReadableContentGuide/ReadableContentGuide.swift:8]: nice, but the second one on the right feels wrong though

File: Sources/StreamVideoSwiftUI/CallingViews/CallingGroupView.swift
Comment (martinmitrevski) [reply Sources/StreamVideoSwiftUI/CallingViews/CallingGroupView.swift]: nice fix!

File: Sources/StreamVideoSwiftUI/Colors.swift
Comment (martinmitrevski) [reply Sources/StreamVideoSwiftUI/Colors.swift:22]: It's a bit of a leftover from the chat SDK, we need to think it through. Best to tackle this in a separate PR.

File: DemoApp/Sources/Views/ReadableContentGuide/ReadableContentGuide.swift
Comment (ipavlidakis) [reply DemoApp/Sources/Views/ReadableContentGuide/ReadableContentGuide.swift:8]: Yeah, the background color of the list is bugging me. I'm looking into it

File: DemoApp/Sources/Views/Reactions/DemoReactionsOverlayView.swift
Comment (martinmitrevski) [reply DemoApp/Sources/Views/Reactions/DemoReactionsOverlayView.swift:28]: if possible, it should be a multiple of 8, or at least even. But leave it up to you.

**Review by martinmitrevski — COMMENTED** [review]
Also, please check if there are any other leftovers from the other repo that could be replaced by the native components.

**Comment by ipavlidakis** [comment]
> Also, please check if there are any other leftovers from the other repo that could be replaced by the native components.

That should be it

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅

**Pull request merged**
